### ISSUE TITLE
schema validation abort directly 

### ISSUE THREAD

**Issue by pint12** [rochacbruno/flasgger#122]
schema validation abort directly 
`    try:    
        jsonschema.validate(data, main_def)            
    except ValidationError as e:       
        abort(Response(str(e), status=400))            
`
should add option to only raise an exception and let the caller to handle it

**Comment by pint12** [comment]
currently i cant catch the err in order to parse it in GUI (the result is not in JSON )

**Comment by allrod5** [comment]
@pint12 `@app.after_request()` annotation may work for this. Not the best solution I know, but it might work.

**Comment by allrod5** [comment]
@pint12 Next release will feature support for using custom validation error handling functions: https://github.com/rochacbruno/flasgger#validation-error-handling

**Comment by pint12** [comment]
0.6.7 ?

**Comment by allrod5** [comment]
@pint12 Yes, originally planned for a major 0.7.0 release but it will no longer be necessary I think. So 0.6.7 will be.

There is, though, a design flaw in _@swag_from_ that I want to fix before releasing 0.6.7:

```
from flasgger import Swagger
from flasgger import swag_from

swagger = Swagger(app, validation_error_handler=my_handler)  # sets default handler

@swag_from('spec.yml', validation=True)  # won't use the default handler
...
```

The same happens if you set a custom default _validation_function_ to _Swagger_.
